autotx/tests/integration/test_swap.py

```diff
@@ -42,9 +43,69 @@ def test_swap(configuration):
     assert new_balance == 0.05
 
 
+def test_swap_recieve_eth(configuration):
+    (user, _, client, _) = configuration
+
+    network_info = SUPPORTED_NETWORKS.get(client.w3.eth.chain_id)
+    eth_address = ETHAddress(network_info.tokens["eth"], client.w3)
+    usdc_address = ETHAddress(network_info.tokens["usdc"], client.w3)
+
+    user_addr = ETHAddress(user.address, client.w3)
+
+    balance = client.w3.eth.get_balance(user_addr.hex)
+    assert int(balance / 10**18) == 9989
+
+    tx = build_swap_transaction(
+        client, 5, eth_address.hex, usdc_address.hex, user_addr.hex, True
+    )
+
+    transaction = user.sign_transaction(
+        {
+            **tx[0].tx,
+            "nonce": client.w3.eth.get_transaction_count(user_addr.hex),
+            "gas": 200000,
+        }
+    )
+
+    hash = client.w3.eth.send_raw_transaction(transaction.rawTransaction)
+
+    receipt = client.w3.eth.wait_for_transaction_receipt(hash)
+
+    print(receipt)
+    if receipt["status"] == 0:
+        print(f"Transaction to swap ETH -> USDC failed ")
+
+    balance = client.w3.eth.get_balance(user_addr.hex)
+    assert int(balance / 10**18) == 9984
+
+    txs = build_swap_transaction(
+        client, 4, usdc_address.hex, eth_address.hex, user_addr.hex, False
+    )
+
+    for i, tx in enumerate(txs):
+        transaction = user.sign_transaction(
+            {
+                **tx.tx,
+                "nonce": client.w3.eth.get_transaction_count(user_addr.hex),
+                "gas": 200000,
```

Comment:
Is the gas limit needed? This should get estimated

Reply:
is needed because when we build the transactions in `build_swap_transaction` we set the gas to `None` - making the transaction execution to fail (this is not the case when executing txs through safe)

Reply:
what about deleting the property?

Reply:
i also tried that and it complains with `TypeError: Transaction must include these fields: {'gas'}` l0l